paddle/fluid/framework/details/op_registry.h

```diff
@@ -156,6 +216,18 @@ struct OpInfoFiller<T, kInplaceOpInference> {
   }
 };
 
+template <typename T>
+struct OpInfoFiller<T, kNoNeedBufferVarsInference> {
+  void operator()(const char* op_type, OpInfo* info) const {
+    info->infer_no_need_buffer_vars_ = [](const VariableNameMap& inputs,
```

Comment:
curious about when will these three parameters be used to get the NoNeedBufferVars, seems now we just return the parameters specified in the macro as an unordered_set?

Reply:
I reserve these parameters to future use. Some ops may not need some forward inputs or outputs when some attribute is true/false. For example, batch_norm_grad_op does not need Bias when use_mkldnn is false.